src/arrayMethodJoin.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let devider = separator;
+    let string = '';
+
+    switch (devider) {
+      case undefined:
+        devider = ',';
+        break;
```

Comment:
For the undefined case, you can just use the default value which you can set in function declaration